docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

```diff
@@ -148,25 +230,59 @@ To set the body content and length of proxied requests, use the `body` property.
 ]
 ```
 
-### Enable token authentication
+### Add a OAuth 2.0 proxy route to your plugin
+
+The data source proxy supports OAuth 2.0 authentication.
 
-To enable token-based authentication for proxied requests, use the `tokenAuth` property.
+Since the request to each route is made server-side, only machine-to-machine authentication is supported. In order words, if you need to use a different grant than client credentials, you need to implement it yourself.
 
-Grafana automatically renews the token when it expires.
+To authenticate using OAuth 2.0, add a `tokenAuth` object to the proxy route definition. If necessary, Grafana performs a request to the URL defined in `tokenAuth` to retrieve a token before making the request to the URL in your proxy route. Grafana automatically renews the token when it expires.
+
+Any parameters defined in `tokenAuth.params` are encoded as `application/x-www-form-urlencoded` and sent to the token URL.
 
 ```json
-"routes": [
-  {
-    "path": "example",
-    "url": "https://api.example.com",
-    "tokenAuth": {
-      "url": "https://login.example.com/oauth2/token",
-      "params": {
-        "grant_type": "client_credentials",
-        "client_id": "{{ .JsonData.clientId }}",
-        "client_secret": "{{ .SecureJsonData.clientSecret }}"
+{
+  "routes": [
+    {
+      "path": "api",
+      "url": "https://api.example.com/v1",
+      "tokenAuth": {
+        "url": "https://api.example.com/v1/oauth/token",
+        "params": {
+          "grant_type": "client_credentials",
+          "client_id": "{{ .SecureJsonData.clientId }}",
+          "client_secret": "{{ .SecureJsonData.clientSecret }}"
+        }
       }
     }
+  ]
+}
+```
+
+## Authenticate using a backend plugin
+
+While the data source proxy supports the most common authentication methods for HTTP APIs, using proxy routes has a few limitations:
+
+- Proxy routes only support HTTP
```

Comment:
Proxy also supports HTTPS, right? I know what you mean here, but I wonder if saying only "HTTP" might lead people to incorrectly think that they need to develop a backend plugin if they want to use https.

Reply:
```suggestion
- Proxy routes only support HTTP/HTTPS
```

Good point! How about this?

Reply:
looks great :)